server/src/main/java/org/apache/iotdb/db/query/expression/unary/ConstantExpression.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.query.expression.unary;
+
+import org.apache.iotdb.db.exception.query.LogicalOptimizeException;
+import org.apache.iotdb.db.exception.query.QueryProcessException;
+import org.apache.iotdb.db.metadata.PartialPath;
+import org.apache.iotdb.db.qp.physical.crud.UDTFPlan;
+import org.apache.iotdb.db.qp.utils.WildcardsRemover;
+import org.apache.iotdb.db.query.expression.Expression;
+import org.apache.iotdb.db.query.udf.core.executor.UDTFExecutor;
+import org.apache.iotdb.db.query.udf.core.layer.ConstantIntermediateLayer;
+import org.apache.iotdb.db.query.udf.core.layer.IntermediateLayer;
+import org.apache.iotdb.db.query.udf.core.layer.LayerMemoryAssigner;
+import org.apache.iotdb.db.query.udf.core.layer.RawQueryInputLayer;
+import org.apache.iotdb.db.utils.CommonUtils;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/** Constant expression */
+public class ConstantExpression extends Expression {
+
+  private final Object value;
+  private final TSDataType dataType;
+
+  public ConstantExpression(TSDataType dataType, String str) throws QueryProcessException {
+    this.dataType = dataType;
+    this.value = CommonUtils.parseValue(dataType, str);
+  }
+
+  public Object getValue() {
+    return value;
+  }
+
+  public TSDataType getDataType() {
+    return dataType;
+  }
+
+  @Override
+  public boolean isPureConstantExpression() {
+    return true;
+  }
+
+  @Override
+  public void concat(List<PartialPath> prefixPaths, List<Expression> resultExpressions) {
+    resultExpressions.add(this);
```

Comment:
This will not handle the following case:

select sin(1 + s1) from root.sg.d1, root.sg.d2

Reply:
I believe the path concat is handled in TimeSeriesOperand, right?

```
create timeseries root.sg1.d1.a with datatype=int32
create timeseries root.sg1.d1.b with datatype=int32
create timeseries root.sg1.d2.a with datatype=int32
create timeseries root.sg1.d2.b with datatype=int32
insert into root.sg1.d1(time, a) values(0, 1)
insert into root.sg1.d1(time, b) values(0, 2)
insert into root.sg1.d2(time, a) values(0, 3)
insert into root.sg1.d2(time, b) values(0, 4)

select a + b - 1 from root.sg1.d1, root.sg1.d2

result:
+-----------------------------+-----------------------------------+-----------------------------------+-----------------------------------+-----------------------------------+
|                         Time|(root.sg1.d1.a + root.sg1.d1.b) - 1|(root.sg1.d1.a + root.sg1.d2.b) - 1|(root.sg1.d2.a + root.sg1.d1.b) - 1|(root.sg1.d2.a + root.sg1.d2.b) - 1|
+-----------------------------+-----------------------------------+-----------------------------------+-----------------------------------+-----------------------------------+
|1970-01-01T08:00:00.000+08:00|                                2.0|                                4.0|                                4.0|                                6.0|
+-----------------------------+-----------------------------------+-----------------------------------+-----------------------------------+-----------------------------------+

```

Reply:
You are right :D